cmd/restic/cmd_init.go

```diff
@@ -33,25 +33,32 @@ func runInit(gopts GlobalOptions, args []string) error {
 		return errors.Fatalf("create repository at %s failed: %v\n", gopts.Repo, err)
 	}
 
-	gopts.password, err = ReadPasswordTwice(gopts,
-		"enter password for new repository: ",
-		"enter password again: ")
-	if err != nil {
-		return err
+	if gopts.MasterKeyFile == "" {
+		gopts.password, err = ReadPasswordTwice(gopts,
+			"enter password for new repository: ",
+			"enter password again: ")
+		if err != nil {
+			return err
+		}
 	}
 
 	s := repository.New(be)
 
-	err = s.Init(gopts.ctx, gopts.password)
+	err = s.Init(gopts.ctx, gopts.MasterKeyFile, gopts.password)
 	if err != nil {
 		return errors.Fatalf("create key in repository at %s failed: %v\n", gopts.Repo, err)
 	}
 
 	Verbosef("created restic repository %v at %s\n", s.Config().ID[:10], gopts.Repo)
 	Verbosef("\n")
-	Verbosef("Please note that knowledge of your password is required to access\n")
-	Verbosef("the repository. Losing your password means that your data is\n")
-	Verbosef("irrecoverably lost.\n")
+	if gopts.MasterKeyFile == "" {
+		Verbosef("Please note that knowledge of your password is required to access\n")
+		Verbosef("the repository. Losing your password means that your data is\n")
+		Verbosef("irrecoverably lost.\n")
+	} else {
+		Verbosef("Please note that you need the masterkey %s to access the repository\n", gopts.MasterKeyFile)
```

Comment:
[nit] Dot is missing at the end of the line.

Reply:
done.